core/storage/ipfs/impl/datastore_leveldb.cpp

```diff
@@ -14,9 +14,9 @@ namespace fc::storage::ipfs {
      * @param value key value to encode
      * @return encoded value as Buffer
      */
-    inline outcome::result<common::Buffer> encode(const CID &value) {
-      OUTCOME_TRY(encoded, value.toBytes());
-      return common::Buffer(std::move(encoded));
+    inline auto encodeKey(const CID &value) {
+      OUTCOME_EXCEPT(encoded, value.toBytes());
```

Comment:
CID's constructed on our side are calculated as hashes, so they are valid and won't throw exception.
Invalid CID's coming from network will not reach here, because they won't be decoded.

Reply:
This should be redone. Need to bring the old code back.